app/controllers/work_packages_controller.rb

```diff
@@ -84,13 +88,25 @@ def show
     end
   end
 
+  def export_dialog
+    respond_with_dialog WorkPackages::Exports::ModalDialogComponent.new(query: @query, project: @project)
+  end
+
   protected
 
+  def load_and_authorize_in_optional_project_for_export
+    load_and_authorize_in_optional_project
+  end
+
+  def load_and_validate_query_for_export
+    load_and_validate_query
+  end
```

Comment:
It’s because I was not able to merge these two controller callbacks into one 
```
before_action :load_and_validate_query, only: :index, unless: -> { request.format.html? }
before_action :load_and_validate_query, only: :export_dialog, if: -> { request.format.html? }
```

but that did not work as calling the same filter with different options is [not supported](https://guides.rubyonrails.org/v7.1.3/action_controller_overview.html#filters).  
So, out of ideas I masked it. 
```
before_action :load_and_validate_query, only: :index, unless: -> { request.format.html? }
before_action :load_and_validate_query_for_export, only: :export_dialog, if: -> { request.format.html? }
```

Happy to do it differently, though.

Reply:
I think you don't really need the 

`if: -> { request.format.html? }`

at the 

`before_action :load_and_authorize_in_optional_project_for_export, only: :export_dialog, if: -> { request.format.html? }` 

before_action declaration, because I wouldn't expect any other format to be requested of the action. But I might be missing something. To me it looks as if it would even be better to have the authorization checked regardless of the format.

In case I am not, then at least this statement and the one for index can simply be merged from:

```
before_action :load_and_authorize_in_optional_project,
              :check_allowed_export,
              :protect_from_unauthorized_export, only: :index
                
before_action :load_and_authorize_in_optional_project_for_export, only: :export_dialog, if: -> { request.format.html? }
```

to

```
before_action :load_and_authorize_in_optional_project,
                only: %i[index export_dialog]
before_action :check_allowed_export,
                :protect_from_unauthorized_export, only: :index
```

But the other one, `load_and_validate_query_for_export`, is admittedly tricky. On that action I again don't think that you need the `if: -> { request.format.html? }` but that does not help a lot.

One could merge the statements again but only at the expense of a very ugly if/unless statement:

```
before_action :load_and_validate_query, only: %i[index export_dialog], if: -> { params[:action] == "index" && !request.format.html? }
```

So there it is probably better to just stick to the masking.